bin/ream/src/main.rs

```diff
@@ -53,11 +53,32 @@ async fn main() {
                 subnets: Subnets::new(),
             };
 
-            let ream_db = ReamDB::new(config.data_dir, config.ephemeral)
+            // database must be clean for checkpoint sync
+            if config.sync_mode.is_checkpoint_sync() {
+                reset_db(config.data_dir.clone()).expect("Unable to delete database");
+            }
+
+            let ream_db = ReamDB::new(config.data_dir.clone(), config.ephemeral)
                 .expect("unable to init Ream Database");
 
             info!("ream database initialized ");
 
+            let _store;
+            let _slot;
+            if config.sync_mode.is_checkpoint_sync() {
+                if config.rpc_url.is_none() {
+                    error!("RPC URL is required for checkpoint sync");
+                }
+                (_store, _slot) = config
+                    .sync_mode
+                    .sync(ream_db.clone(), &config.rpc_url.unwrap())
+                    .await
+                    .map_err(|err| panic!("{err}"))
+                    .unwrap();
+
+                info!("Checkpoint Sync complete");
+            }
```

Comment:
```suggestion
            if database.is_empty {
                 initialize_database_using_checkpoint
            }
            
            // todo add syncing code once p2p code is more mature.
```

We aren't actually syncing here. this should just check if the database is empty and if it is, it should download the State ,Block, and Blobs. So that we can then sync from P2P